cmd/ocm/create/machinepool/cmd.go

```diff
@@ -175,72 +231,90 @@ func run(cmd *cobra.Command, argv []string) error {
 		}
 	}
 
-	// Create the client for the OCM API:
-	connection, err := ocm.NewConnection().Build()
+	machineTypeList, err := machineTypeListGetter.GetMachineTypeOptions(cluster)
 	if err != nil {
-		return fmt.Errorf("Failed to create OCM connection: %v", err)
+		return err
 	}
-	defer connection.Close()
-
-	// Get the client for the cluster management api
-	clusterCollection := connection.ClustersMgmt().V1().Clusters()
-
-	cluster, err := c.GetCluster(connection, clusterKey)
-	if err != nil {
-		return fmt.Errorf("Failed to get cluster '%s': %v", clusterKey, err)
+	if err := flags.CheckOneOf("instance-type", machineTypeList); err != nil {
+		return err
 	}
 
-	if cluster.State() != cmv1.ClusterStateReady {
-		return fmt.Errorf("Cluster '%s' is not yet ready", clusterKey)
+	if args.AvailabilityZone != "" {
+		if cluster.CloudProviderId() != c.ProviderGCP {
+			return fmt.Errorf(
+				"At this time, OCM CLI does not support setting 'availability-zone'"+
+					" for clusters using the cloud provider '%s'",
+				cluster.CloudProviderId())
+		}
 	}
 
-	machineTypeList, err := provider.GetMachineTypeOptions(connection.ClustersMgmt().V1(),
-		cluster.CloudProvider().ID(),
-		cluster.CCS().Enabled())
-	if err != nil {
-		return err
-	}
-	err = arguments.CheckOneOf(cmd.Flags(), "instance-type", machineTypeList)
-	if err != nil {
-		return err
+	return nil
+}
+
+func buildMachinePool(
+	machinePoolId string,
+) (*cmv1.MachinePool, error) {
+	labels := make(map[string]string)
+	if args.Labels != "" {
+		for _, label := range strings.Split(args.Labels, ",") {
+			tokens := strings.Split(label, "=")
+			labels[strings.TrimSpace(tokens[0])] = strings.TrimSpace(tokens[1])
+		}
 	}
 
+	taintBuilders := []*cmv1.TaintBuilder{}
+	if args.Taints != "" {
+		for _, taint := range strings.Split(args.Taints, ",") {
+			tokens := strings.FieldsFunc(taint, arguments.Split)
+			taintBuilders = append(taintBuilders, cmv1.NewTaint().Key(tokens[0]).Value(tokens[1]).Effect(tokens[2]))
+		}
+	}
 	mpBuilder := cmv1.NewMachinePool().
-		ID(machinePoolID).
-		InstanceType(args.instanceType).
+		ID(machinePoolId).
+		InstanceType(args.InstanceType).
 		Labels(labels).
 		Taints(taintBuilders...)
 
-	if len(args.additionalSecurityGroupIds) != 0 {
-		for i, sg := range args.additionalSecurityGroupIds {
-			args.additionalSecurityGroupIds[i] = strings.TrimSpace(sg)
+	if len(args.AdditionalSecurityGroupIds) != 0 {
+		for i, sg := range args.AdditionalSecurityGroupIds {
+			args.AdditionalSecurityGroupIds[i] = strings.TrimSpace(sg)
 		}
 		mpBuilder.AWS(
```

Comment:
This looks like is missing a provider check. I know this wasn't introduced in this PR but if its not something user should be setting for GCP or should be setting it only for AWS, it would be good to add that check as well.